### PR TITLE
fix: [DEL-5884]: Moving  secret manager config away from redis 

### PR DESCRIPTION
**Pull request by gaurav-jain1**
## Describe your changes
Few months back, we introduced a cache (backed up by redis) to store secret manager config. This is essential for the pipeline which includes lot many secret to decrypt, and hence acquire call takes too much time (most of the case it times out). To reduce latencies, we cache the secret manger config, to avoid network latencies.
Since now these configs are in redis, which stores this data in plain text in its storage file system, hence there can be security concern.
To address this issue, making this cache local to manager pod, so that such config doesn't goes out of manager process.

Testing:
<Haven't started yet, will do it before merge.>

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43309)

### CODE REVIEW

**Review by raghuAtWings — DISMISSED** [review]
> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43309)* status: :shipit: complete! ✓

File: 400-rest/src/main/java/software/wings/expression/NgSecretManagerFunctor.java [hunk 1]
```diff
@@ -140,11 +140,14 @@ private Object obtainInternal(String secretIdentifier) {
     List<EncryptedDataDetail> encryptedDataDetails = null;
     if (secretsCache != null) {
       delegateMetricsService.recordDelegateMetricsPerAccount(accountId, SECRETS_CACHE_LOOKUPS);
-      EncryptedDataDetails cachedValue = secretsCache.get(String.valueOf(keyHash));
+      EncryptedDataDetails cachedValue = secretsCache.getIfPresent(String.valueOf(keyHash));
       if (cachedValue != null) {
         // Cache hit.
+        log.info("cache hit for {}", String.valueOf(keyHash));
```
Comment (ankit-harness) [inline 400-rest/src/main/java/software/wings/expression/NgSecretManagerFunctor.java:146]: Is this logging for debugging? We should not put this in Prod

**Review by mdmilic — COMMENTED** [review]
> Reviewed 4 of 6 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43309)* status: 0 of 1 approvals obtained.

**Review by gaurav-jain1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43309)* status: 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/expression/NgSecretManagerFunctor.java` line 146 at r2](https://reviewable.io/reviews/harness/harness-core/43309#-NN7mDC6xjuzoGA_BWiw-r2-146:-NNL_I4Y9V7ql90B4-pP:bfll3yh) ([raw file](https://github.com/harness/harness-core/blob/c10a00dd89c14dde0a82e59e198a4fabc11979a3/400-rest/src/main/java/software/wings/expression/NgSecretManagerFunctor.java#L146)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Is this logging for debugging? We should not put this in Prod
</blockquote></details>

yes, this logging is to test in PR env, reverted now.

**Review by xingchi-jin — APPROVED** [review]
> Reviewed 4 of 6 files at r1, 2 of 2 files at r2, 8 of 8 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43309)* status: Some of the checks are too old. 1. unresolved discussions.

**Pull request merged**